azure-pipelines.yml

```diff
@@ -2,32 +2,55 @@
 # Create and test a Python package on multiple Python versions.
 # Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
 # https://docs.microsoft.com/azure/devops/pipelines/languages/python
-
+pr: none
 trigger:
-- master
-
-pool:
-  vmImage: 'ubuntu-latest'
-strategy:
-  matrix:
-    Python36:
-      python.version: '3.6'
-    Python37:
-      python.version: '3.7'
-    Python38:
-      python.version: '3.8'
-steps:
-- task: UsePythonVersion@0
-  inputs:
-    versionSpec: '$(python.version)'
-  displayName: 'Use Python $(python.version)'
+  branches:
+    include:
+    - '*'
 
-- script: |
-    python -m pip install --upgrade pip
-    pip install -r requirements.txt
-  displayName: 'Install dependencies'
+stages:
+- stage: static_checks
+  displayName: Static Checks
+  dependsOn: []
+  jobs:
+    - job:
+      displayName: "Check Git Tags"
+      steps:
+      - script: echo "##vso[build.addbuildtag]Automated"
+        condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
 
-- script: |
-    pip install pytest pytest-azurepipelines
-    pytest
-  displayName: 'pytest'
+- stage: test_and_build
+  displayName: 'Test and build'
+  dependsOn: []
+  jobs:
+    - job:
+      displayName: ' '
+      strategy:
+        matrix:
+          python36:
+            PYTHON_VERSION: '3.6'
+          python37:
+            PYTHON_VERSION: '3.7'
+          python38:
+            PYTHON_VERSION: '3.8'
+      pool:
+        vmImage: 'ubuntu-18.04'
+      steps:
+        - task: UsePythonVersion@0
+          inputs:
+            versionSpec: $(PYTHON_VERSION)
+        - script: |
+            pip install -U pip
+            pip install -U -r requirements.txt
+          displayName: 'Install requirements'
+        - script: |
+            pytest
+          displayName: 'Tests'
+#        - bash: |
+#            pip install -U pip codecov
+#            report_name="ubuntu18-04_$(PYTHON_VERSION)"
+#            python -m codecov --token $(CODECOV_TOKEN) \
+#              --file coverageReport.xml \
+#              -n ${report_name}
+#          displayName: 'Send coverage to codecov'
+#          condition: succeededOrFailed()
```

Comment:
To make this work you'll need to add a `CODECOV_TOKEN` to Azure as a secure variable. You can get the token from https://codecov.io/gh/data61/anonlink-client, then you can add it in Azure following these [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch).

I've just added the token as a secure environment variable now, but it would be worth checking you have permission to get the token and add it in Azure DevOps.

Reply:
Yep this will be fixed in a separate issue